src/api/notify-namespace.ts

```diff
@@ -0,0 +1,538 @@
+        },
+        method,
+        data
+      }
+    );
+  }
+
+  /**
+   * Create a new {@link MinedTransactionWebhook} to track mined transactions
+   * sent by the app associated with the app id.
+   *
+   * Note that the webhook will be created in the app network of the provided app id.
+   *
+   * @param url The URL that the webhook should send events to.
+   * @param type The type of webhook to create.
+   * @param params Parameters object containing the app id.
+   */
+  createWebhook(
+    url: string,
+    type: WebhookType.MINED_TRANSACTION,
+    params: TransactionWebhookParams
+  ): Promise<MinedTransactionWebhook>;
+
+  /**
+   * Create a new {@link DroppedTransactionWebhook} to track dropped transactions
+   * sent by the app associated with the app id.
+   *
+   * Note that the webhook will be created in the app network of the provided app id.
+   *
+   * @param url The URL that the webhook should send events to.
+   * @param type The type of webhook to create.
+   * @param params Parameters object containing the app id.
+   */
+  createWebhook(
+    url: string,
+    type: WebhookType.DROPPED_TRANSACTION,
+    params: TransactionWebhookParams
+  ): Promise<DroppedTransactionWebhook>;
+
+  /**
+   * Create a new {@link NftActivityWebhook} to track NFT transfers.
+   *
+   * @param url The URL that the webhook should send events to.
+   * @param type The type of webhook to create.
+   * @param params Parameters object containing the NFTs to track and the
+   *   network the webhook should be created on.
+   */
+  createWebhook(
+    url: string,
+    type: WebhookType.NFT_ACTIVITY,
+    params: NftWebhookParams
+  ): Promise<NftActivityWebhook>;
+
+  /**
+   * Create a new {@link AddressActivityWebhook} to track address activity.
+   *
+   * @param url The URL that the webhook should send events to.
+   * @param type The type of webhook to create.
+   * @param params Parameters object containing the addresses to track and the
+   *   network the webhook should be created on.
+   */
+  createWebhook(
+    url: string,
+    type: WebhookType.ADDRESS_ACTIVITY,
+    params: AddressWebhookParams
+  ): Promise<AddressActivityWebhook>;
+  async createWebhook(
+    url: string,
+    type: WebhookType,
+    params: NftWebhookParams | AddressWebhookParams | TransactionWebhookParams
+  ): Promise<
+    | MinedTransactionWebhook
+    | DroppedTransactionWebhook
+    | NftActivityWebhook
+    | AddressActivityWebhook
+  > {
+    let appId;
+    if (
+      type == WebhookType.MINED_TRANSACTION ||
+      type === WebhookType.DROPPED_TRANSACTION
+    ) {
+      if (!('appId' in params)) {
+        throw new Error('Transaction Webhooks require an app id.');
+      }
+      appId = params.appId;
+    }
+
+    let network = NETWORK_TO_WEBHOOK_NETWORK.get(this.config.network);
+    let filters;
+    let addresses;
+    if (type === WebhookType.NFT_ACTIVITY) {
+      if (!('filters' in params) || params.filters.length === 0) {
```

Comment:
If you want, you can write this condition as
```ts
if (!params.filters?.length)
```
but very reasonable if you think that's too tricky.

Similarly, the condition in the next branch could be written
```ts
if (!params?.addresses?.length)
```

Reply:
I originally had it that way, but typescript can't recognize the overloads for some reason and will throw errors since it thinks that `params` can be any of the different param types. This then requires a cast every time we access params. Not sure if it's because I'm missing something.

Reply:
Oh huh, you're right. I didn't realize that, good thinking!